benchmark/opperf/utils/benchmark_operators_pytest.py

```diff
@@ -0,0 +1,110 @@
+                           ((60,60,60,60), {"numHidden": 60}),
+                           ((90,90,90,90), {"numHidden": 90}),],
+    'batch_dot'         : [((10,10,10), {"matrix1": (20,30), "matrix2": (30,40)}),
+                           ((20,20,20), {"matrix1": (40,50), "matrix2": (50,60)}),
+                           ((40,40,40), {"matrix1": (60,70), "matrix2": (70,80)})]
+}
+
+def generate_test_cases():
+    tests = []
+    for op_name, cases in test_cases.items():
+        for case in cases:
+            tests.append((op_name, case[0], case[1]))
+    return tests
+
+def generate_test_ids():
+    test_ids = []
+    for op_name, cases in test_cases.items():
+        for case in cases:
+            s = op_name + "-shape_"
+            for i in range(len(case[0])):
+                s += str(case[0][i])
+                if (i != len(case[0])-1):
+                    s += "x"
+            params = case[1].items()
+            if len(params) != 0:
+                s += "-params"
+                for key, value in params:
+                    s += "_" + str(key) + "_"
+                    if isinstance(value, tuple):
+                        for i in range(len(value)):
+                            s += str(value[i])
+                            if (i != len(value)-1):
+                                s += "x"
+                    else:
+                        s += str(value)
+            test_ids.append(s)
+    return test_ids
+
+generate_inputs = {
+    'reshape'               : lambda shape, metadata: {'newshape': metadata["newshape"], 'shape': metadata["newshape"]},
+    'swapaxes'              : lambda shape, metadata: {"axis1": metadata["axis1"], "axis2": metadata["axis2"],
+                                                       "dim1": metadata["axis1"], "dim2": metadata["axis2"]},
+    'activation'            : lambda shape, metadata: {'act_type': metadata["actType"]},
+    'batch_norm'            : lambda shape, metadata: {'gamma': (shape[1],), 'beta': (shape[1],), 'running_mean': (shape[1],), 'running_var': (shape[1],),
+                                                       'moving_mean': (shape[1],), 'moving_var': (shape[1],)},
+    'convolution'           : lambda shape, metadata: {"weight": (metadata["numFilter"], shape[1]) + metadata["kernel"], "kernel": metadata["kernel"],
+                                                       "bias": (metadata["numFilter"],), "num_filter": metadata["numFilter"]},
+    'masked_softmax'        : lambda shape, metadata: {"mask": mx.np.array(round(mx.np.random.rand(*shape)), dtype="bool")},
+    'fully_connected'       : lambda shape, metadata: {"weight": (metadata["numHidden"], shape[-1]), "bias": (metadata["numHidden"],), 
+                                                       "num_hidden": metadata["numHidden"], "flatten": False},
+    'batch_dot'             : lambda shape, metadata: {'lhs': shape + metadata["matrix1"], 'a': shape + metadata["matrix1"],
```

Comment:
What is 'a' and 'b' for batch dot?

Reply:
Different parameters names in .npx module (lhs, rhs in .nd).